### PR TITLE
19 Card Digit Support Part 2

### PR DESCRIPTION
**Pull request by csabol-stripe**
## Summary
Enables auto-advancing when able and loading indicator to STPPaymentCardTextField

## Motivation

Refinement and improvement of behavior for >16 digit card numbers

## Testing

Tested in sample apps

### CODE REVIEW

File: Stripe/STPAnalyticsClient.m [hunk 1]
```diff
@@ -321,6 +321,19 @@ - (void)log3DS2ChallengeFlowCompletedWithConfiguration:(STPPaymentConfiguration
     [self logPayload:payload];
 }
 
+#pragma mark - Card Metadata
+
+- (void)logUserEnteredCompletePANBeforeMetadataLoadedWithConfiguration:(STPPaymentConfiguration *)configuration {
+    NSDictionary *configurationDictionary = [self.class serializeConfiguration:configuration];
+    NSMutableDictionary *payload = [self.class commonPayload];
+    [payload addEntriesFromDictionary:@{
+                                        @"event": @"stripeios.card_metadata_loaded_too_slow",
```
Comment (csabol-stripe) [inline Stripe/STPAnalyticsClient.m:330]: open to event name suggestions here cc @mshafrir-stripe

File: Stripe/STPPaymentCardTextField.m [hunk 2]
```diff
@@ -1317,15 +1322,44 @@ - (void)formTextFieldTextDidChange:(STPFormTextField *)formTextField {
...
                 }
-                // no auto-advance
-            }];
+            } else {
+                [self.viewModel validationStateForCardNumberWithHandler:^(STPCardValidationState state) {
+                    if ([self.viewModel.cardNumber isEqualToString:number]) {
+                        [self updateCVCPlaceholder];
+                        self.cvcField.validText = [self.viewModel validationStateForCVC] != STPCardValidationStateInvalid;
+                        formTextField.validText = state != STPCardValidationStateInvalid;
+                        if (state == STPCardValidationStateValid) {
```
Comment (csabol-stripe) [inline Stripe/STPPaymentCardTextField.m:1343]: maybe we should log if it's invalid too?

File: Stripe/STPPaymentCardTextField.m [hunk 3]
```diff
@@ -1340,6 +1340,10 @@ - (void)formTextFieldTextDidChange:(STPFormTextField *)formTextField {
                         [self updateCVCPlaceholder];
                         self.cvcField.validText = [self.viewModel validationStateForCVC] != STPCardValidationStateInvalid;
                         formTextField.validText = state != STPCardValidationStateInvalid;
+                        if (state == STPCardValidationStateValid) {
+                            // log that user entered full complete PAN before we got a network response
+                            [[STPAnalyticsClient sharedClient] logUserEnteredCompletePANBeforeMetadataLoadedWithConfiguration:[STPPaymentConfiguration sharedConfiguration]];
```
Comment (davidme-stripe) [inline Stripe/STPPaymentCardTextField.m:1345]: Wouldn't this log on paste?

Comment (csabol-stripe) [reply Stripe/STPPaymentCardTextField.m:1345]: yeah. I think that's a legitimate log. There might be an optimization to split out cases

**Pull request merged**